### PR TITLE
Fix Metrics interface

### PR DESCRIPTION
**Pull request by topicus**
- Fix Metrics struct
- Use snapshot before get Gauge value
- Fix Gauge type during marshalling

### CODE REVIEW

File: metrics/metrics.go [hunk 1]
```diff
@@ -69,6 +69,7 @@ type Metrics interface {
 	MeasureBackend5xx(t time.Time)
 	IncErrorsStreaming(routeId string)
 	RegisterHandler(path string, handler *http.ServeMux)
+	UpdateGauge(key string, value float64)
```
Comment (szuecs) [inline metrics/metrics.go]: this will break prometheus provider, please implement a dummy in the prometheus provider, too

Comment (topicus) [reply metrics/metrics.go]: Good catch!

Comment (topicus) [reply metrics/metrics.go]: Sorry I got confused. It is already implemented https://github.com/zalando/skipper/blob/master/metrics/prometheus.go#L297

File: metrics/codahale.go [hunk 2]
```diff
@@ -291,10 +291,15 @@ func (sm skipperMetrics) MarshalJSON() ([]byte, error) {
 	for name, metric := range sm {
 		values := make(map[string]interface{})
 		var metricsFamily string
+
 		switch m := metric.(type) {
 		case metrics.Gauge:
 			metricsFamily = "gauges"
 			values["value"] = m.Value()
+		case metrics.GaugeFloat64:
```
Comment (addityasingh) [inline metrics/codahale.go]: Does it make sense to also add a test similar to this https://github.com/zalando/skipper/blob/master/metrics/prometheus_test.go#L695?

**Comment by szuecs** [comment]
:+1:

File: metrics/codahale.go (hunk 2 above)
Comment (szuecs) [reply metrics/codahale.go]: It’s not needed as long as you implemented the interface and you did, so it is fine.
Thanks

**Comment by aryszka** [comment]
:+1:

**Pull request merged**